### ISSUE TITLE
Frontend validation for create leads form

### ISSUE THREAD

**Issue by galbwe** [codefordenver/partner-finder#145]
Depends on #143 being merged

Add the following validation to the create leads form. Feel free to add error messages in this issue or leave them for another PR.

[ ] assigned field must be the username of a registered user
[ ] company name field cannot be null or blank
[ ] email field must be a properly formatted email address (don't worry about checking that it is a real email yet)
[ ] phone field must be in the format `(###) ###-####` or `###-####`
[ ] website, linkedin, twitter, facebook, and instagram fields must be properly formatted urls
